TCAPractice/ContentView.swift

```diff
@@ -1,17 +1,65 @@
 import SwiftUI
+import TCAPracticeCore
+import ComposableArchitecture
 
 struct ContentView: View {
+
+  let store: StoreOf<QuizFeatureReducer>
+
     var body: some View {
-        VStack {
-            Image(systemName: "globe")
-                .imageScale(.large)
-                .foregroundStyle(.tint)
-            Text("Hello, world!")
-        }
-        .padding()
+      WithViewStore(store, observe: {$0}) { viewStore in
+        GeometryReader(content: { geometry in
+          cardView(viewStore)
+            .offset(x: viewStore.offset)
+            .gesture(
+              DragGesture()
+                .onChanged({ gesture in
+                  viewStore.send(.changeOffset(gesture.translation.width))
+                })
+                .onEnded({ _ in
+                  if viewStore.offset < -geometry.size.width * 0.5 {
+                    viewStore.send(.swipe(.left, -geometry.size.width))
+                  } else if viewStore.offset > geometry.size.width * 0.5 {
+                    viewStore.send(.swipe(.right, geometry.size.width))
+                  }
+                })
```

Comment:
Let me preface this by saying that I'm not sure what the recommended approach for this is in TCA, but I don't love the fact that this is broken up across two actions. Might it be better if the viewStore accepted a single action that takes a gesture and the geometry size? That way viewStore could determine the resulting action directly. Combining them could, in theory, also make it more testable by being able to test partial swipes, etc. (Albeit, taking away the swipeLeft/Right actions from being directly testable.)

Reply:
good point! just updated it